src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.ts

```diff
@@ -13,20 +13,24 @@ export class DotEditPageWorkflowsActionsComponent implements OnInit {
     @Input() inode: string;
     @Input() label: string;
 
-    workflowsActions: Observable<MenuItem[]>;
+    workflowsMenuActions: Observable<MenuItem[]>;
 
     constructor(private workflowsService: DotWorkflowService) {}
 
     ngOnInit() {
-        this.workflowsActions = this.workflowsService
-            .getContentWorkflowActions(this.inode)
-            .map((workflows: DotWorkflowAction[]) => this.getWorkflowOptions(workflows));
+        this.workflowsMenuActions = this.workflowsService.getContentWorkflowActions(this.inode).map((workflows: DotWorkflowAction[]) => {
+            return this.getWorkflowOptions(workflows);
+        });
     }
 
     private getWorkflowOptions(workflows: DotWorkflowAction[]): MenuItem[] {
         return workflows.map((workflow: DotWorkflowAction) => {
             return {
-                label: workflow.name
+                label: workflow.name,
+                command: () => {
+                    const request: Observable<any[]> = this.workflowsService.updateWorkflowActions(this.inode, workflow.id);
+                    request.subscribe();
```

Comment:
As far as I understand how Worflows actions work, after you trigger one, you need to update the ```workflowsMenuActions``` becuase they can change if you move to another step of the workflow by fire the actions.

Does this fire endpoint return something? maybe the new actions to fire? Maybe @jdotcms can give us more information here.

Also, what happen is there is an error in the request here? you need to catch that.

I created this service for that: https://github.com/dotCMS/core-web/blob/master/src/app/api/services/dot-http-error-manager/dot-http-error-manager.service.ts

Here is an use case: https://github.com/dotCMS/core-web/blob/ddc1f0f76b31fdacf26777463f665aa1f74b25ea/src/app/portlets/content-types/edit/content-types-edit-resolver.service.ts#L41

Reply:
I think the end point return a contentelet (I don't know why), so we should call this method again:

https://github.com/dotCMS/core-web/blob/827ffcdcf8dad3364eb2aef1976bd77efce471d6/src/app/api/services/dot-workflow/dot-workflow.service.ts#L39

Also we should print some message in the GlobalMessage